plugins/woocommerce-admin/client/products/product-validation.ts

```diff
@@ -51,6 +56,20 @@ export const validate = (
 		);
 	}
 
+	if ( values.sku && values.sku.length > 0 && ! isValidSlug( values.sku ) ) {
```

Comment:
I think we can group these 2 conditions like
```javascript
if ( values.sku ) {
    if ( values.sku.length > 0 && ! isValidSlug( values.sku ) ) { ... }
    if ( values.sku.length > 160 ) { ... }
}
```

Reply:
In general, we try to avoid nesting like this.  It only shortens the line by a few characters but reduces readability.  The same applies for our SCSS selectors where we try not to go more than 3 levels deep (though this is not a firm rule).

Reply:
You could use `values.sku?.length`.

Reply:
Shouldn't we check if it is a valid SKU, not a valid slug? Even if the current implementation has those being the exact same checks, it would be good to make an explicit check whether it is a valid SKU, since then in the future, if the validation of the two differed, this code would still be correct.

Reply:
>You could use values.sku?.length.

Good call!  I'll add this.

>Shouldn't we check if it is a valid SKU, not a valid slug

I'm wary of doing this because there is no such thing as a valid SKU in the eyes of WooCommerce.  We've changed the behavior here and we are in fact checking against the "slug rules."  

Should those two rules separate in the future, I think it would be worth renaming, but I think it would be dishonest to name this `isValidSku` when that does not have its own set of rules yet.

Reply:
> In general, we try to avoid nesting like this. It only shortens the line by a few characters but reduces readability. The same applies for our SCSS selectors where we try not to go more than 3 levels deep (though this is not a firm rule).
No it's not a firm rule, it's a fact! I agree excessive nesting can be hard to read and understand but nesting by a context with not to much levels helps a lot to find and understand things. For example in this case we are talking about all validations that are related to the sku.